core/common/src/main/java/alluxio/Configuration.java

```diff
@@ -493,58 +494,45 @@ private static void checkWorkerPorts() {
   }
 
   /**
-   * {@link PropertyKey#USER_FILE_BUFFER_BYTES} should not bigger than {@link Integer#MAX_VALUE}
-   * bytes.
+   * Validates user file buffer size configuration.
    *
-   * @throws IllegalArgumentException if USER_FILE_BUFFER_BYTES bigger than Integer.MAX_VALUE
+   * @throws IllegalStateException if invalid user file buffer size configuration is encountered
    */
   private static void checkUserFileBufferBytes() {
     if (!containsKey(PropertyKey.USER_FILE_BUFFER_BYTES)) { // load from hadoop conf
       return;
     }
     long usrFileBufferBytes = getBytes(PropertyKey.USER_FILE_BUFFER_BYTES);
-    Preconditions.checkArgument((usrFileBufferBytes & Integer.MAX_VALUE) == usrFileBufferBytes,
-        PreconditionMessage.INVALID_USER_FILE_BUFFER_BYTES.toString(), usrFileBufferBytes);
+    Preconditions.checkState((usrFileBufferBytes & Integer.MAX_VALUE) == usrFileBufferBytes,
+        PreconditionMessage.INVALID_USER_FILE_BUFFER_BYTES.toString(),
+        PropertyKey.Name.USER_FILE_BUFFER_BYTES, usrFileBufferBytes);
   }
 
   /**
    * Validates Zookeeper-related configuration and prints warnings for possible sources of error.
+   *
+   * @throws IllegalStateException if invalid Zookeeper configuration is encountered
    */
   private static void checkZkConfiguration() {
-    if (getBoolean(PropertyKey.ZOOKEEPER_ENABLED)) {
-      Preconditions.checkState(containsKey(PropertyKey.ZOOKEEPER_ADDRESS),
-          PreconditionMessage.ERR_ZK_ADDRESS_NOT_SET.toString(),
-          PropertyKey.ZOOKEEPER_ADDRESS.toString());
-    } else if (containsKey(PropertyKey.ZOOKEEPER_ADDRESS)) {
-      LOG.warn("{} is configured, but {} is set to false", PropertyKey.ZOOKEEPER_ADDRESS.toString(),
-          PropertyKey.ZOOKEEPER_ENABLED.toString());
-    }
-  }
-
-  /**
-   * Checks that the configuration values are reasonable.
-   */
-  private static void checkConfigurationValues() {
-    checkWorkerPorts();
-    checkUserFileBufferBytes();
-    checkZkConfiguration();
+    Preconditions.checkState(
+        containsKey(PropertyKey.ZOOKEEPER_ADDRESS) == containsKey(PropertyKey.ZOOKEEPER_ADDRESS),
```

Comment:
I think it's reasonable for a user to set `zookeeper.enabled=false` but not set an address. Should this instead be `checkState(getBoolean(PropertyKey.ZOOKEEPER_ENABLED) == containsKey(PropertyKey.ZOOKEEPER_ADDRESS))`?